### DISCUSSION TITLE
Nilan Combi 302 + Waveshare

### DISCUSSION THREAD

**Discussion by jamespreedy** [veista/nilan#73] [General]
Hi folks. I'm planning to start using this integration and have just ordered my Waveshare RS485 TO ETH (B) to use with my Combi 302. I had a couple of questions if someone could please point me in the right direction.

I use a Waveshare already with my Solax solar inverter, and I needed to spend some time configuring it via its web page to get it up and running - I've included a screenshot below. Are there any particular settings I need to have to get the Nilan talking to Home Assistant correctly?

![Screenshot 2023-08-04 at 8 31 57 pm](https://github.com/veista/nilan/assets/559531/8126756a-ae3d-408f-a243-1697b79b3a04)

Secondly, I had a question about the Modbus cabling. At the moment the Combi has both a touch screen and the gateway connected via Modbus. Cables from each both seem to be connected into the same terminals on the MVHR itself. My plan is to remove the gateway and connect the Waveshare in its place. Is that correct, is there anything I should be aware of? My thinking is to take a bunch of photos and if I can't connect the MVHR to Home Assistant, I should at least be able to get it back to using the Nilan gateway again.

Thanks in advance, really appreciate it.

**Comment by veista** [comment]
Hi,

Parity should be even. The calbling should be separate from the display, but yes it replaces the nilan gateway.

**Comment by jamespreedy** [comment]
@veista - thanks for the reply, and thanks for your amazing work on this integration! 🍺
I disconnected the gateway, and checked that the touch screen is still working - it is. Phew.

I've connected 485A (blue) and 485B (blue white) on the Waveshare to ports 6 (blue) and 7 (blue white) on the Nilan - it has the 8 port connector on it. I've also tried swapping the blue and white/blue cable - just in case it made a difference.

When I boot everything up I don't get the blue connected light on the Waveshare (not sure if I should yet).
I try to add the device in Home Assistant, select Nilan => TCP Configuration and enter the IP and port. Is unit ID 30? In the service settings on the touch screen, the Modbus ID is 30. Assume this should be the same?

I get an error saying "Unsupported Device".

I have a Nilan Combi 302 Polar with a CTS602 HMI touch panel.
When I go to Unit Information on the touch panel, it says I have 

Unit Type - COMBI 300 N
Software Version - 1.6.25.0
Panel Software - 1.6.24.0

I noticed that in one of the other discussions someone posted their screenshot of their Waveshare config.
They had instruction time out as 192, and enable multi-host as yes - I've tried these and it didn't seem to make any difference.

I wanted to try to get some debug logs, how do I do that?
I added "nilan: debug" to my logger config, but didn't see anything in the logs.

If you have a moment, do you have any advice on what else I could try please?

![Screenshot 2023-08-06 at 1 44 55 pm](https://github.com/veista/nilan/assets/559531/3e58cb08-de18-476a-adb1-fe443a7cfd49)

![IMG_3638](https://github.com/veista/nilan/assets/559531/0d36c0b3-e884-4bc1-aa61-4fba9f14402e)

![Screenshot 2023-08-06 at 1 49 40 pm](https://github.com/veista/nilan/assets/559531/9b24bc26-2e79-48a2-935f-afc913e4cb56)

![IMG_3639](https://github.com/veista/nilan/assets/559531/fbb83cdb-0aee-4032-a008-c26f61332e80)

**Reply by veista** [reply]
Hi. Unsupported device means it is working. Follow the instructions on the front page! Combi 300 n is not yet supported!
